src/routes/conversation/[id]/web-search/+server.ts

```diff
@@ -73,45 +63,63 @@ export async function GET({ params, locals, url }) {
 
 				appendUpdate("Searching Google", [webSearch.searchQuery]);
 				const results = await searchWeb(webSearch.searchQuery);
-
-				let text = "";
 				webSearch.results =
 					(results.organic_results &&
-						results.organic_results.map((el: { link: string }) => el.link)) ??
+						results.organic_results.map((el: { title: string; link: string }) => {
+							const { title, link } = el;
+							const { hostname } = new URL(link);
+							return { title, link, hostname };
+						})) ??
 					[];
-
-				if (results.answer_box) {
-					// if google returns an answer box, we use it
-					webSearch.answerBox = JSON.stringify(removeLinks(results.answer_box));
-					text = webSearch.answerBox;
-					appendUpdate("Found a Google answer box");
-				} else if (results.knowledge_graph) {
-					// if google returns a knowledge graph, we use it
-					webSearch.knowledgeGraph = JSON.stringify(removeLinks(results.knowledge_graph));
-					text = webSearch.knowledgeGraph;
-					appendUpdate("Found a Google knowledge page");
-				} else if (webSearch.results.length > 0) {
-					let tries = 0;
-
-					while (!text && tries < 3) {
-						const searchUrl = webSearch.results[tries];
-						appendUpdate("Browsing result", [JSON.stringify(searchUrl)]);
+				webSearch.results = webSearch.results
+					.filter(({ link }) => !link.includes("youtube.com")) // filter out youtube links
+					.slice(0, MAX_N_PAGES_SCRAPE); // limit to first 10 links only
+
+				let paragraphChunks: { source: WebSearchSource; text: string }[] = [];
+				if (webSearch.results.length > 0) {
+					appendUpdate("Browsing results");
+					const promises = webSearch.results.map(async (result) => {
+						const { link } = result;
+						let text = "";
 						try {
-							text = await parseWeb(searchUrl);
-							if (!text) throw new Error("text of the webpage is null");
+							text = await parseWeb(link);
+							appendUpdate(`Browsing webpage: ${link}`);
```

Comment:
```suggestion
							appendUpdate("Browsing webpage", [link]);
```

Fairly small change but it puts the url as an argument below 

![image](https://github.com/huggingface/chat-ui/assets/25119303/abcc1ec5-2ad0-4eea-be58-68ee9a75484a)

Reply:
yes nice catch

Reply:
I've removed those messages altogether https://github.com/huggingface/chat-ui/pull/427/commits/dd155791604bcce58749cab32086164cf7d43e0e as it is very noisy and we show sources at the end. wdyt?